Cannot compile library due to _BSD_SOURCE wrong usage

Hello, 
I'm not a C expert, but I'm having problem compiling when I include [open62541](https://github.com/open62541/open62541/tree/master) lib (the amalgamated single file)

The error I have is 
```
/mnt/c/Program Files/Dassault Systemes/B426_Cloud/win_b64/resources/Dymola/source/ModelicaInternal.c: In function ‘ModelicaInternal_fullPathName’:
/mnt/c/Program Files/Dassault Systemes/B426_Cloud/win_b64/resources/Dymola/source/ModelicaInternal.c:609:37: error: operator '||' has no left operand
```
line 609 looks like that in my dymola install :
`#if defined(_WIN32) || (_BSD_SOURCE || _XOPEN_SOURCE >= 500 || _XOPEN_SOURCE && _XOPEN_SOURCE_EXTENDED || (_POSIX_VERSION >= 200112L))
`
The concerned line in the actual lib seems this one : https://github.com/modelica/ModelicaStandardLibrary/blob/bc5c458731def2ad5b128feb1b7e9a563b031a6a/Modelica/Resources/C-Sources/ModelicaInternal.c#L687

I think this occur because the _BSD_SOURCE define is empty because of this line in my lib :
https://github.com/open62541/open62541/blob/6115a47125d7e6ccd77488b8572807171adb0602/tools/ua-tool/ua.c#L18

open62541 seems legit to define _BSD_SOURCE empty 
see https://ftp.gnu.org/old-gnu/Manuals/glibc-2.2.3/html_node/libc_13.html

I would suggest replacing _BSD_SOURCE by defined(_BSD_SOURCE) in ModelicaInternal.c
I can open a MR is it seem the right fix 

Note that there is three occurence of _BSD_SOURCE to fix

--- a/Modelica/Resources/C-Sources/ModelicaInternal.c
+++ b/Modelica/Resources/C-Sources/ModelicaInternal.c
@@ -671,6 +671,13 @@ Modelica_ERROR:
 
 _Ret_z_ const char* ModelicaInternal_fullPathName(_In_z_ const char* name) {
     /* Get full path name of file or directory */
+#undef MODELICA_INTERNAL_HAVE_POSIX_REALPATH
+#if defined(_BSD_SOURCE) || \
+    (defined(_XOPEN_SOURCE) && _XOPEN_SOURCE >= 500) || \
+    (defined(_XOPEN_SOURCE) && defined(_XOPEN_SOURCE_EXTENDED)) || \
+    _POSIX_VERSION >= 200112L
+#define MODELICA_INTERNAL_HAVE_POSIX_REALPATH
+#endif
 
 #if defined(_WIN32)
     char* fullName;
@@ -684,7 +691,7 @@ _Ret_z_ const char* ModelicaInternal_fullPathName(_In_z_ const char* name) {
     fullName = ModelicaDuplicateString(tempName);
     ModelicaConvertToUnixDirectorySeparator(fullName);
     return fullName;
-#elif (_BSD_SOURCE || _XOPEN_SOURCE >= 500 || _XOPEN_SOURCE && _XOPEN_SOURCE_EXTENDED || _POSIX_VERSION >= 200112L)
+#elif defined(MODELICA_INTERNAL_HAVE_POSIX_REALPATH)
     char* fullName;
     char localbuf[BUFFER_LENGTH];
     size_t len;
@@ -710,10 +717,10 @@ _Ret_z_ const char* ModelicaInternal_fullPathName(_In_z_ const char* name) {
     return fullName;
 #endif
 
-#if (_BSD_SOURCE || _XOPEN_SOURCE >= 500 || _XOPEN_SOURCE && _XOPEN_SOURCE_EXTENDED || _POSIX_VERSION >= 200112L)
+#if defined(MODELICA_INTERNAL_HAVE_POSIX_REALPATH)
 FALLBACK_getcwd:
 #endif
-#if (_BSD_SOURCE || _XOPEN_SOURCE >= 500 || _XOPEN_SOURCE && _XOPEN_SOURCE_EXTENDED || _POSIX_VERSION >= 200112L || _POSIX_)
+#if defined(MODELICA_INTERNAL_HAVE_POSIX_REALPATH) || defined(_POSIX_)
     {
         /* No such system call in _POSIX_ available (except realpath for existing paths) */
         char* cwd = getcwd(localbuf, sizeof(localbuf));
@@ -734,6 +741,7 @@ FALLBACK_getcwd:
     }
     return fullName;
 #endif
+#undef MODELICA_INTERNAL_HAVE_POSIX_REALPATH
 }
 
 _Ret_z_ const char* ModelicaInternal_temporaryFileName(void) {

